### PR TITLE
(WIP) Begin UI dark theme

### PR DESCRIPTION
**Pull request by nullpixel**
Let's make it more "limitless" https://cloud.githubusercontent.com/assets/2975241/20806609/3d06d346-b7c1-11e6-9ebe-3a540105f9d1.gif

TODO:

- [x] Make all pages dark
- [ ] New bar icons

Screenshots:

simulator screen shot 3 dec 2016 15 28 46
simulator screen shot 3 dec 2016 15 28 48
simulator screen shot 3 dec 2016 15 28 50
simulator screen shot 3 dec 2016 15 28 51
simulator screen shot 3 dec 2016 15 28 52

### CODE REVIEW

**Comment by JohnCoates** [comment]
Looking good! We've got a dark mode concept in #28 you can take a look at also. I suggest you start wrapping dark mode into a feature flag, because it's unlikely to get merged otherwise.

**Comment by nullpixel** [comment]
@JohnCoates Yeah, that's where I was getting inspiration from.

File: Source/UI/Packages/View Controllers/SearchController.mm [hunk 1]
```diff
@@ -123,9 +123,12 @@ - (id) initWithDatabase:(Database *)database query:(NSString *)query {
             textField = [search_ searchField];
         else
             textField = MSHookIvar<UITextField *>(search_, "_searchField");
-        [textField setBackgroundColor:[UIColor cydia_black]];
-        [textField setTextColor:[UIColor whiteColor]];
-        [textField setKeyboardAppearance:UIKeyboardAppearanceDark];
+        //Dark Mode check
```
Comment (JohnCoates) [inline Source/UI/Packages/View Controllers/SearchController.mm]: No need for this comment, code below is self explanatory.

File: Source/Application/Application.mm [hunk 2]
```diff
@@ -339,6 +351,7 @@ - (void)setUpViewControllers {
     NSMutableArray *controllers([NSMutableArray array]);
     for (UITabBarItem *item in items) {
         UINavigationController *controller([[[UINavigationController alloc] init] autorelease]);
+        // [[controller navigationBar] setTitleTextAttributes:@{NSForegroundColorAttributeName: [UIColor whiteColor]}];
```
Comment (JohnCoates) [inline Source/Application/Application.mm]: Not being used? Let's have it removed

File: Source/UI/Sources/View Controllers/SourcesController.mm [hunk 3]
```diff
@@ -100,6 +101,10 @@ - (UITableViewCell *) tableView:(UITableView *)tableView cellForRowAtIndexPath:(
 		[cell addGestureRecognizer:favouriteGesture];
 	}
 	
+    
+    // TODO: fix background shit
```
Comment (JohnCoates) [inline Source/UI/Sources/View Controllers/SourcesController.mm]: Please remove swearing.

File: Source/UI/Sources/View Controllers/SourcesController.mm [hunk 4]
```diff
@@ -400,6 +405,8 @@ - (void) updateButtonsForEditingStatusAnimated:(BOOL)animated {
 - (void) loadView {
     list_ = [[[UITableView alloc] initWithFrame:[[UIScreen mainScreen] applicationFrame]
                                           style:UITableViewStylePlain] autorelease];
+    // TODO: fix background shit
```
Comment (JohnCoates) [inline Source/UI/Sources/View Controllers/SourcesController.mm]: Please remove swearing.

**Review by JohnCoates — CHANGES_REQUESTED** [review]
Made some comments on some changes before merging, otherwise looks good!

File: Source/UI/Sources/View Controllers/SourcesController.mm (hunk 3 above)
Comment (nullpixel) [reply Source/UI/Sources/View Controllers/SourcesController.mm]: Yeah, sorry, left it to navigate around code

**Pull request merged**